graphql-parser/src/lexer.rs

```diff
@@ -2,14 +2,23 @@
 use std::str::CharIndices;
 use std::iter::Peekable;
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(PartialEq, Debug)]
 pub enum Token<'a> {
-  Eof,
+  Eof(usize, usize),
+  Punctuator(Punctuator, usize),
```

Comment:
The issue is that the use case will end up as `Punctuator(Punctuator, start, start)` which means we have a whole variable doing nothing for most of the time, except in the one case it is spread. 

Currently it still sort of reads correctly as it is written like `Punctuator(Spread(start), end)` there is just one extra paren. I prefer it this way, but if @cksac also wants both then that will be it.